doc/en/developer/tutorial-api.md

```diff
@@ -0,0 +1,1116 @@
+        # with a specific instance
+        $self->{app_metrics}->{ $entry->{component} }->{display} = $entry->{component};
+        $self->{app_metrics}->{ $entry->{component} }->{connections} = $entry->{value}
+    };
+    
+     # Exactly the same thing with errors
+    foreach my $entry (@{ $results->{errors} }) {
+        # Don't need to redefine the display key, just assign a value to the error key_value while
+        # keeping the $entry->{component} key to associate the value with the good instance
+        $self->{app_metrics}->{ $entry->{component} }->{errors} = $entry->{value};
+    };
+}
+```
+
+### 5.4 New file : api.pm
+
+As explained in the previous section, the custom file will contain all needed functions about input and the way to process it.
+
+This new file needs to contains the packages and libraries declarations :
+
+```perl
+package apps::myawesomeapp::api::custom::api;
+
+use strict;
+use warnings;
+
+use centreon::plugins::http;
+use centreon::plugins::templates::catalog_functions qw(catalog_status_threshold_ng);
+use JSON::XS;
+```
+
+It also contains the following functions :
+* new constructor : construct the object in the same way than in mode file previously
+* set_options
+* set_defaults
+* check_options
+* settings
+* request_api
+
+#### new constructor
+
+```perl
+sub new {
+    my ($class, %options) = @_;
+    my $self  = {};
+    bless $self, $class;
+
+    # Check if an output option is available
+    if (!defined($options{output})) {
+        print "Class Custom: Need to specify 'output' argument.\n";
+        exit 3;
+    }
+    # Check if options are avaliable
+    if (!defined($options{options})) {
+        $options{output}->add_option_msg(short_msg => "Class Custom: Need to specify 'options' argument.");
+        $options{output}->option_exit();
+    }
+
+    if (!defined($options{noptions})) {
+        # Adding options legacy from appsmetrics.pm in single mode
+        $options{options}->add_options(arguments => {
+            'hostname:s'           => { name => 'hostname' },
+            'proto:s'              => { name => 'proto', default => 'https' },
+            'port:s'               => { name => 'port', default => 443 },
+            'timeout:s'            => { name => 'timeout' },
+            'unknown-status:s'     => { name => 'unknown_status', default => '%{http_code} < 200 or %{http_code} >= 300' },
+            'warning-status:s'     => { name => 'warning_status' },
+            'critical-status:s'    => { name => 'critical_status', default => '' }
+        });
+    }
+    # Adding Help structure to the object
+    $options{options}->add_help(package => __PACKAGE__, sections => 'REST API OPTIONS', once => 1);
+    # Adding output structure to the object
+    $self->{output} = $options{output};
+    # Command line legacy from appsmetrics.pm in single mode
+    $self->{http} = centreon::plugins::http->new(%options);
+
+    return $self;
+}
+```
+
+#### set_options
```

Comment:
This section should have more explanations about different functions but at this time some part remains unclear for me.